components/financial_assistance/app/domain/financial_assistance/operations/applications/haven/add_eligibility_determination.rb

```diff
@@ -87,17 +87,19 @@ def build_or_update_applicants_eligibility_determinations(verified_family, _prim
           # rubocop:enable Metrics/CyclomaticComplexity
 
           def update_eligibility_determinations(vthh, eligibility_determination)
-            verified_eligibility_determination = vthh.eligibility_determinations.max_by(&:determination_date) #Finding the right Eligilbilty Determination
-            source = "Faa"
-            verified_aptc = verified_eligibility_determination.maximum_aptc.to_f > 0.00 ? verified_eligibility_determination.maximum_aptc : 0.00
-            eligibility_determination.update_attributes(
-              max_aptc: verified_aptc,
-              determined_at: verified_eligibility_determination.determination_date,
-              aptc_csr_annual_household_income: verified_eligibility_determination.aptc_csr_annual_household_income,
-              aptc_annual_income_limit: verified_eligibility_determination.aptc_annual_income_limit,
-              csr_annual_income_limit: verified_eligibility_determination.csr_annual_income_limit,
-              source: source
-            )
+            ed_update_params = { source: 'Faa', determined_at: DateTime.now }
+
+            if vthh.eligibility_determinations.present?
+              verified_eligibility_determination = vthh.eligibility_determinations.max_by(&:determination_date)
+              verified_aptc = verified_eligibility_determination.maximum_aptc.to_f > 0.00 ? verified_eligibility_determination.maximum_aptc : 0.00
+              ed_update_params.merge!({ max_aptc: verified_aptc,
+                                        determined_at: verified_eligibility_determination.determination_date,
+                                        aptc_csr_annual_household_income: verified_eligibility_determination.aptc_csr_annual_household_income,
+                                        aptc_annual_income_limit: verified_eligibility_determination.aptc_annual_income_limit,
+                                        csr_annual_income_limit: verified_eligibility_determination.csr_annual_income_limit })
+            end
+
+            eligibility_determination.update_attributes!(ed_update_params)
```

Comment:
Updating eligibility_determination irrespective of eligibility determination presence in the response payload. Updates additional fields only when there is ed present in the response.